### PR TITLE
Unconditionally remove proxy headers to prevent httpoxy

### LINKED ISSUE #14516: HTTPoxy in haproxy
Perhaps it would be ideal to add the patch for https://httpoxy.org/ in the default haproxy template?

### PR DESCRIPTION
**Pull request by simo5**
See https://httpoxy.org/ for more details.
Fixes #14516

### CODE REVIEW

File: images/router/haproxy/conf/haproxy-config.template [hunk 1]
```diff
@@ -171,7 +171,10 @@ frontend public
   {{- if (eq .StatsPort -1) }}
   monitor-uri /_______internal_router_healthz
   {{- end }}
-  
+
+  # Strip off Proxy headers to prevent HTTTpoxy (https://httpoxy.org/)
```
Comment (mrogers950) [inline images/router/haproxy/conf/haproxy-config.template]: HTTTpoxy -> HTTPoxy

**Comment by enj** [comment]
@openshift/networking

**Comment by simo5** [comment]
[test]

**Comment by knobunc** [comment]
Ok, let's [merge] this now and we can decide if we need to add control of removing arbitrary headers in the future.

**Pull request merged**